### ISSUE TITLE
Jitter bigger than Delay causes a (random) failure at runtime

### ISSUE THREAD

**Issue by JanecekPetr** [jhalterman/failsafe#115]
Consider this:

```
RetryPolicy retryPolicy = new RetryPolicy()
    .retryOn(IllegalArgumentException.class)
    .withDelay(100, TimeUnit.MILLISECONDS)
    .withJitter(200, TimeUnit.MILLISECONDS);

Failsafe.with(retryPolicy)
    .onRetry((result, failure, context) -> System.out.println("retried"))
    .run(() -> {
        throw new IllegalArgumentException("nope");
    });
```

This will generally retry a few times and then fail with `java.lang.IllegalArgumentException: timeout value is negative`.

This is obviously because the jitter is larger than delay, and therefore (delay - jitter) can be negative. It would be nice if in such cases the delay was simply capped to zero.

What I wanted, in fact, was a random delay between 0 and 10 seconds, and 
```
RetryPolicy retryPolicy = new RetryPolicy()
    .retryOn(() -> someCheck())
    .withJitter(10, TimeUnit.SECONDS);
```
is apparently the wrong thing to do. On the other hand, I find it much more readable than the correct alternative:

```
RetryPolicy retryPolicy = new RetryPolicy()
    .retryOn(() -> someCheck())
    .withDelay(5, TimeUnit.SECONDS)
    .withJitter(5, TimeUnit.SECONDS);
```

**Comment by jhalterman** [comment]
Would be happy to accept a PR for this. IMO, we should throw IllegalArgumentException if a Jitter value is provided that is bigger than the delay, or a delay that is smaller than the jitter.

**Comment by jhalterman** [comment]
#129 is released in 1.1.0.
